cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/SystemDatasetRuntimeModule.java

```diff
@@ -88,6 +88,11 @@ protected void configure() {
         bind(DatasetDefinition.class)
           .annotatedWith(Names.named(Constants.Dataset.TABLE_TYPE)).to(InMemoryTableDefinition.class);
 
+        bind(String.class)
+          .annotatedWith(Names.named(Constants.Dataset.TABLE_TYPE_NO_TX)).toInstance("table-no-tx");
+        bind(DatasetDefinition.class)
+          .annotatedWith(Names.named(Constants.Dataset.TABLE_TYPE_NO_TX)).to(InMemoryMetricsTableDefinition.class);
```

Comment:
I just notice that since I last looked at this, the following few lines were added, they actually already bind this to a tpe literal. Maybe you can use that? I guess we should follow a common pattern...

Reply:
I did see that binding, but I prefer this approach since it makes it clear that we are binding MetricsTable to be used as a non-transactional table type.

Reply:
ok makes sense